apps/web/src/lib/functions/file-loaders/epub/generate-epub-html.ts

```diff
@@ -139,32 +139,48 @@ export default function generateEpubHtml(
       htmlHref = itemIdToHtmlRef[itemIdRef];
     }
 
-    const regexResult = /.*<body(?:[^>]*id="(?<id>.+?)")*[^>]*>(?<body>(.|\s)+)<\/body>.*/.exec(
-      data[htmlHref] as string
-    )!;
+    let parsedContent = parser.parseFromString(data[htmlHref] as string, 'text/html');
+    let body = parsedContent.body;
 
-    const bodyId = regexResult?.groups?.id || '';
-    let innerHtml = regexResult?.groups?.body || '';
+    if (!body?.childElementCount) {
+      parsedContent = parser.parseFromString(data[htmlHref] as string, 'text/xml');
+      body = parsedContent.querySelector('body'); // XMLDocument doesn't seem to have the body property
```

Comment:
the actual documents are still expected to be in html form - it is just interpreted with a different parser. Typings also still have a body property so i think its safe to assume that the body should be defined and if not that it would be anyways an error / unsupported format

Reply:
When I tested this in my browser console (Firefox):
```js
p = new DOMParser()
x = p.parseFromString('<html><body><p>hello world</p></body></html>', 'text/xml')
x.body
```
I got `null`. `XMLDocument`s having the `body` attribute defined doesn't seem to be a behavior we can rely on.

Reply:
Tested it in Chromium, `x.body` is `null` there as well.

Reply:
because you are using literal xml while epubs are using xhtml. the body comes from the respective namespace.

so e. g. 
```js
p = new DOMParser()
x = p.parseFromString('<html xmlns="http://www.w3.org/1999/xhtml"><body><p>hello world</p></body></html>', 'text/xml')
x.body
```

works fine. but its ok - either i change it with some other update or i'll just keep it as is.. should not have noticeable runtime difference anyways

Reply:
Ahhh, that makes sense, thanks for the explanation!